arbitrator/tools/stylus_benchmark/src/benchmark.rs

```diff
@@ -0,0 +1,119 @@
+// Copyright 2021-2024, Offchain Labs, Inc.
+// For license information, see https://github.com/OffchainLabs/nitro/blob/master/LICENSE
+
+use arbutil::evm::{api::Ink, EvmData};
+use core::time::Duration;
+use jit::machine::WasmEnv;
+use jit::program::{
+    exec_program, get_last_msg, pop_with_wasm_env, start_program_with_wasm_env, JitConfig,
+};
+use prover::programs::{config::CompileConfig, config::PricingParams, prelude::StylusConfig};
+use std::path::PathBuf;
+use std::str;
+use stylus::native::compile;
+use wasmer::Target;
+
+const EVM_API_METHOD_REQ_OFFSET: u32 = 0x10000000;
+
+const NUMBER_OF_BENCHMARK_RUNS: u32 = 7;
+const NUMBER_OF_TOP_AND_BOTTOM_RUNS_TO_DISCARD: u32 = 2;
+
+fn check_result(req_type: u32, req_data: &Vec<u8>) {
+    let _ = match str::from_utf8(req_data) {
+        Ok(v) => v,
+        Err(e) => panic!("Invalid UTF-8 sequence: {}", e),
+    };
+
+    match req_type {
+        0 => return,
+        1 => panic!("ErrExecutionReverted user revert"),
+        2 => panic!("ErrExecutionReverted user failure"),
+        3 => panic!("ErrOutOfGas user out of ink"),
+        4 => panic!("ErrDepth user out of stack"),
+        _ => panic!("ErrExecutionReverted user unknown"),
+    }
+}
+
+fn run(compiled_module: Vec<u8>) -> (Duration, Ink) {
+    let calldata = Vec::from([0u8; 32]);
+    let evm_data = EvmData::default();
+    let config = JitConfig {
+        stylus: StylusConfig {
+            version: 2,
+            max_depth: 10000,
+            pricing: PricingParams { ink_price: 1 },
+        },
+        compile: CompileConfig::version(2, true),
+    };
+
+    let exec = &mut WasmEnv::default();
+
+    let module = exec_program(
```

Comment:
Is this code benchmarking only the JIT execution? I'm worried about the differences between the JIT and the interpreter, so it would be nice to benchmark non-JIT execution as well.

Reply:
That is a good point.
By interpreter you mean execution through the prover's binary?

@tsahee, any takes on that?

Reply:
In the prover the bottleneck is not how long it takes the prover to execute but how many steps it takes - because that affects how long it will take to hash for BOLD.
And in stylus, that's a very different constraint.